meta-mbl-distro/scripts/create-update-payload

```diff
@@ -239,108 +286,131 @@ def get_files_for_boot_part(deploy_dir, tinfoil=None):
     return boot_files
 
 
-def copy_files_to_subdir(paths, dst_dir):
+def get_archived_file_spec_for_rootfs(image_name, tinfoil):
+    """
+    Get an ArchivedFileSpec for the root partition image.
     """
-    Copy files to a destination directory, renaming them when specified.
+    deploy_dir = get_bitbake_conf_var("DEPLOY_DIR_IMAGE", tinfoil)
+    if not deploy_dir:
+        bb.fatal("Failed to get value of DEPLOY_DIR_IMAGE variable")
```

Comment:
I'm wondering if we should move this error checking to the `get_bitbake_conf_var` function? IIRC I think it was previously relying on some bitbake variables being `None` because of the way rootfs and app updates were plumbed in. Maybe we can just fail now if any bitbake variable is `None` - but I'm not sure.

Reply:
Good point. I'll take a look.

Reply:
Factored out in https://github.com/ARMmbed/meta-mbl/pull/730/commits/7613c6bf408aa394e97ff010094b7102dbb46ae1.

I had to keep an option to not fail when the variable isn't set for when we're checking the `_SKIP` variables. While I was in that code, I also factored out the code to check whether a partition is skipped.